python/pyspark/sql/readwriter.py

```diff
@@ -177,31 +180,35 @@ def json(self, path, schema=None):
         :param path: string represents path to the JSON dataset,
                      or RDD of Strings storing JSON objects.
         :param schema: an optional :class:`StructType` for the input schema.
+        :param primitivesAsString: infers all primitive values as a string type. If None is set, \
+                                   it uses the default value, ``false``.
+        :param prefersDecimal: infers all floating-point values as a decimal type. If the values \
+                               do not fit in decimal, then it infers them as doubles. If None is \
+                               set, it uses the default value, ``false``.
+        :param allowComments: ignores Java/C++ style comment in JSON records. If None is set, \
+                              it uses the default value, ``false``.
+        :param allowUnquotedFieldNames: allows unquoted JSON field names. If None is set, \
+                                        it uses the default value, ``false``.
+        :param allowSingleQuotes: allows single quotes in addition to double quotes. If None is \
+                                        set, it uses the default value, ``true``.
+        :param allowNumericLeadingZero: allows leading zeros in numbers (e.g. 00012). If None is \
+                                        set, it uses the default value, ``false``.
+        :param allowBackslashEscapingAnyCharacter: allows accepting quoting of all character \
+                                                   using backslash quoting mechanism. If None is \
+                                                   set, it uses the default value, ``false``.
+        :param columnNameOfCorruptRecord: allows renaming the new field having malformed string \
+                                          created by ``PERMISSIVE`` mode. This overrides \
+                                          ``spark.sql.columnNameOfCorruptRecord``. If None is set, \
+                                          it uses the default value ``_corrupt_record``.
+        :param mode: allows a mode for dealing with corrupt records during parsing. If None is \
```

Comment:
I think I should move this option to the right pleace too. Let me push a commit more tomorrow.